From ad65770512d2baeb45f5d0622d985f9856b7cc1e Mon Sep 17 00:00:00 2001
From: Richard Liu <r66033@freescale.com>
Date: Thu, 10 Mar 2016 17:53:59 +0800
Subject: [PATCH] MA-7715 fix GPU kernel panic reported by android CTS

The patch removes the dependence between cache flush operation and node.
Node is not used anymore when flush cache. Cache flush can work with only
logical address passed into underlying functions does not need physical
address.

Signed-off-by: Richard Liu <r66033@freescale.com>
---
 drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel.c | 34 ++------------------------
 1 file changed, 2 insertions(+), 32 deletions(-)

diff --git a/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel.c b/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel.c
index 2c8b2f9..6258f1f 100644
--- a/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel.c
+++ b/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel.c
@@ -2,7 +2,7 @@
 *
 *    The MIT License (MIT)
 *
-*    Copyright (c) 2014 - 2015 Vivante Corporation
+*    Copyright (c) 2014 - 2016 Vivante Corporation
 *
 *    Permission is hereby granted, free of charge, to any person obtaining a
 *    copy of this software and associated documentation files (the "Software"),
@@ -26,7 +26,7 @@
 *
 *    The GPL License (GPL)
 *
-*    Copyright (C) 2014 - 2015 Vivante Corporation
+*    Copyright (C) 2014 - 2016 Vivante Corporation
 *
 *    This program is free software; you can redistribute it and/or
 *    modify it under the terms of the GNU General Public License
@@ -1611,7 +1611,6 @@ gckKERNEL_Dispatch(
 #endif
     gckVIRTUAL_COMMAND_BUFFER_PTR buffer;
 
-    gckVIDMEM_NODE nodeObject;
     gctBOOL powerMutexAcquired = gcvFALSE;
 
     gcmkHEADER_ARG("Kernel=0x%x FromUser=%d Interface=0x%x",
@@ -2417,35 +2416,6 @@ gckKERNEL_Dispatch(
     case gcvHAL_CACHE:
 
         logical = gcmUINT64_TO_PTR(Interface->u.Cache.logical);
-
-        if (Interface->u.Cache.node)
-        {
-            gcmkONERROR(gckVIDMEM_HANDLE_Lookup(
-                Kernel,
-                processID,
-                Interface->u.Cache.node,
-                &nodeObject));
-
-            /* Verify the nodeObject. */
-            if ((nodeObject == gcvNULL)
-            ||  (nodeObject->node == gcvNULL)
-            ||  (nodeObject->node->VidMem.memory == gcvNULL)
-            )
-            {
-                /* Invalid object. */
-                gcmkONERROR(gcvSTATUS_INVALID_OBJECT);
-            }
-
-            if (nodeObject->node->VidMem.memory->object.type == gcvOBJ_VIDMEM
-             || nodeObject->node->Virtual.contiguous
-            )
-            {
-                /* If memory is contiguous, get physical address. */
-                gcmkONERROR(gckOS_UserLogicalToPhysical(
-                    Kernel->os, logical, &paddr));
-            }
-        }
-
         bytes = (gctSIZE_T) Interface->u.Cache.bytes;
         switch(Interface->u.Cache.operation)
         {
-- 
1.8.0

